### PR TITLE
Gdb debug server

### PR DESCRIPTION
**Pull request by jfhs**
This is very basic implementation of GDB remote stub, that allows any debugger that supports GDB protocol and ppc64 architecture to connect to rpcs3 and debug threads (currently only PPU threads are supported).

Even though there is a debugger and memory viewer in rpcs3 itself, they are not the most comfortable tools to work with. My initial goal was to make debugging possible from IdaPro, but even though it has GDB debugger, it seems to have wrong register information for ppc64 and so doesn't work. I am planning to make Ida debugger plugin. Having that would simplify finding complex bugs which require some disassembly and debugging. At the moment I have tested gdb and radare2, both work, but radare2 require some configs.

I understand that only small fraction of users would need that, so whole feature is disabled by default, and can be enabled by defining `WITH_GDB_DEBUGGER`.

I am not proficient in c++/rpcs3, so if someone experienced have time, please take a look and tell me if there is anything that should be fixed. In particular I am not sure if I use `idm` properly, and do I need those `hex_to_uxx` functions, or there is something already available (I haven't found anything)?

Overall adding this should have no impact on codebase, there are only 3-5 lines added in Emu core, but can give some better tools for debugging.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -107,6 +109,13 @@ void Emulator::Init()
 	fs::create_path(dev_hdd1);
 	fs::create_path(dev_usb);
 
+#ifdef WITH_GDB_DEBUGGER
+	if (g_gdb_debugger_id) {
+		idm::remove<GDBDebugServer>(g_gdb_debugger_id);
```
Comment (Nekotekina) [inline rpcs3/Emu/System.cpp]: There is `fxm` for single object

File: rpcs3/Emu/Cell/PPUThread.cpp [hunk 2]
```diff
@@ -392,13 +429,20 @@ void ppu_thread::exec_task()
 
 	while (true)
 	{
+#ifdef WITH_GDB_DEBUGGER
+		if (test(state, cpu_flag::dbg_pause)) {
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/PPUThread.cpp]: Need to move this out of loop into the `check_state` body

**Comment by Nekotekina** [comment]
I'll try to resolve issues manually later.

**Comment by jfhs** [comment]
Sorry, I didn't have time yet to fix this. I hope to have time later this week. Do you have any other comments on code? Because fixing what you mentioned would be fast.

**Comment by jfhs** [comment]
I've fixed issues you mentioned and rebased to latest commit. I don't understand why AppVeyor failed, looking into log I don't see any errors.

**Comment by paulsapps** [comment]
I think there is a time limit on the compile, perhaps it timed out?

**Comment by Nekotekina** [comment]
Must be ok to merge.

**Pull request merged**